core-rust/core-api-server/src/core_api/conversions/lts.rs

```diff
@@ -54,117 +59,59 @@ pub fn to_api_lts_committed_transaction_outcome(
-            }
-        }
+    let mut fee_balance_changes = index_map_new();
+    let records = database.batch_get_ancestry(fee_payments.keys());
+    for (paid_fee_amount_xrd, ancestry) in fee_payments.values().zip(records) {
+        let ancestry = ancestry.expect("a vault must be owned by an account");
+        let account_address = GlobalAddress::new_or_panic(ancestry.root.0 .0);
+        fee_balance_changes
+            .entry(account_address)
+            .or_insert_with(Decimal::zero)
+            .sub_assign(*paid_fee_amount_xrd);
     }
 
-    let (assumed_fee_payer, assumed_fee_balance_change) = match (
-        exact_fee_debit,
-        biggest_xrd_debit,
-        total_fee_balance_change == net_xrd_balance_change,
-    ) {
-        // If an entity debited the exact fee - it's probably that entity
-        // - This covers the case where entity X paid the fee but didn't otherwise transfer XRD
-        (Some(entity_address), _, true) => (Some(entity_address), total_fee_balance_change),
-        // Else use the entity that debitted the most XRD - who is most likely to be the fee payer
-        // - This is accurate in the case where someone transferred XRD from their account and paid the fee
-        (None, Some((_, entity_address)), true) => (Some(entity_address), total_fee_balance_change),
-        // If there's been no XRD debit, ot it doesn't equal the total fee, then we should be at genesis or in an end of
-        // epoch scenario without a fee payer
-        _ => (None, Decimal::ZERO),
-    };
-
-    balance_changes
-        .iter()
-        .map(|(entity_address, resource_changes)| {
-            let changes = if assumed_fee_payer == Some(*entity_address) {
-                models::LtsEntityFungibleBalanceChanges {
-                    entity_address: to_api_global_address(context, entity_address)?,
-                    fee_balance_change: Some(Box::new(models::LtsFungibleResourceBalanceChange {
-                        resource_address: to_api_resource_address(context, &RADIX_TOKEN)?,
-                        balance_change: to_api_decimal(&assumed_fee_balance_change),
-                    })),
-                    non_fee_balance_changes: resource_changes
-                        .iter()
-                        .filter_map(|(resource_address, balance_change)| {
-                            if *resource_address == RADIX_TOKEN {
-                                let fungible_balance_change =
-                                    get_fungible_balance(balance_change).unwrap();
-                                let non_fee_balance_change =
-                                    fungible_balance_change - assumed_fee_balance_change;
-                                if non_fee_balance_change == Decimal::ZERO {
-                                    return None;
-                                }
-                                return Some(to_api_lts_fungible_resource_balance_change(
+    Ok(balance_changes
```

Comment:
I don't believe this is quite right. If a fee_balance_change is offset by an equal and opposite transfer in, then the balance maybe won't be in the `balance_changes`.

I would suggest the simplest is to make `non_fee_balance_changes = balance_changes` and add the `paid_fee_amount_xrd` XRD fees to the `non_fee_balance_changes` alongside line 72.

The logic below then also becomes a lot simpler - basically just filtering out 0s and mapping.

Reply:
So I had to refactor all of this as part of pulling through damson/elm, so I suggest probably wanting to close this PR and put these changes on top of the elm PR

Reply:
ok, let me do it 👍 (I assume "the elm PR" == https://github.com/radixdlt/babylon-node/pull/529)